### PR TITLE
Custom xml consolidation

### PR DESCRIPTION
**Pull request by abelal**
- Unifies the logic for creating custom.xml.
- Minor cleanups and clarifications.

JIRA Ticket: SB-8614.

### CODE REVIEW

**Comment by abelal** [comment]
@Noor-Ahsan, please review.

File: scripts/toaster-layers-recipes-config.py [hunk 1]
```diff
@@ -25,27 +23,129 @@ def get_layer_name(lpath):
...
+        machine_found = True
+    if distro_found and machine_found:
+        break
+lc.close()
 
-layer_and_recipes = OrderedDict()
-
-recipeListFile = builddir + "/conf/recipes-list.txt"
 customXML = builddir + "/custom.xml"
+fd = open(customXML, 'a')
```
Comment (sharidas) [inline scripts/toaster-layers-recipes-config.py]: Wouldn't it be nice if we make it as write mode instead of append mode?

Comment (sharidas) [inline scripts/toaster-layers-recipes-config.py]: The change looks pretty neat to me. And I appreciate that you took the effort to consolidate most of the work into the python file. Thanks for the work @abelal  :)

Comment (abelal) [inline scripts/toaster-layers-recipes-config.py]: @sharidas, very valid and would make it more clear where the file is actually created at first. I'll make the change.

**Pull request closed without merging**